e2e-tests/spring-rest-mongo/src/test/java/com/foo/spring/rest/mongo/SpringRestMongoController.java

```diff
@@ -0,0 +1,115 @@
+package com.foo.spring.rest.mongo;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.ServerAddress;
+import org.evomaster.client.java.controller.EmbeddedSutController;
+import org.evomaster.client.java.controller.api.dto.AuthenticationDto;
+import org.evomaster.client.java.controller.api.dto.SutInfoDto;
+import org.evomaster.client.java.controller.problem.ProblemInfo;
+import org.evomaster.client.java.controller.problem.RestProblem;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.testcontainers.containers.GenericContainer;
+
+import java.sql.Connection;
+import java.util.List;
+import java.util.Map;
+
+public abstract class SpringRestMongoController extends EmbeddedSutController {
+
+    private final Class<?> applicationClass;
+
+    private ConfigurableApplicationContext ctx;
+    private MongoClient mongoClient;
+
+    public SpringRestMongoController(Class<?> applicationClass) {
+        this.applicationClass = applicationClass;
+    }
+
+    private final static int MONGO_PORT = 27017; //https://docs.mongodb.com/manual/reference/default-mongodb-port/
+
+    private final GenericContainer mongodb = new GenericContainer("mongo:3.2")
```

Comment:
the current version of Mongo is 4.2.2. Any specific reason to use 3.2?

Reply:
It was the version used in OCVN